scripts/cc-cli/src/utils/balance.ts

```diff
@@ -27,33 +31,62 @@ export function readAmountFromHex(amount: string): BN {
   return new BN(amount.slice(2), 16);
 }
 
-export interface Balance {
-  free: BN;
-  reserved: BN;
-  miscFrozen: BN;
-  feeFrozen: BN;
+export interface AccountBalance {
+  address: string;
+  transferable: BN;
+  locked: BN;
+  bonded: BN;
+  total: BN;
 }
 
 export async function getBalance(address: string, api: any) {
   const account = await api.query.system.account(address);
-  return balanceFromData(account.data);
+  return balanceFromData(account.data, address);
 }
 
-function balanceFromData(data: any): Balance {
+function balanceFromData(data: any, address: string): AccountBalance {
   return {
-    free: data.free,
-    reserved: data.reserved,
-    miscFrozen: data.miscFrozen,
-    feeFrozen: data.feeFrozen,
+    address,
+    transferable: data.free.sub(data.miscFrozen),
+    bonded: data.miscFrozen,
+    locked: data.reserved,
+    total: data.free,
   };
 }
 
-export function printBalance(balance: Balance) {
-  console.log("Available:", toCTCString(balance.free.sub(balance.miscFrozen)));
-  console.log("Free:", toCTCString(balance.free));
-  console.log("Reserved:", toCTCString(balance.reserved));
-  console.log("Misc Frozen:", toCTCString(balance.miscFrozen));
-  console.log("Fee Frozen:", toCTCString(balance.feeFrozen));
```

Comment:
Q: I see 5 fields removed, 4 fields added. Which one is missing ?

Reply:
Its the other way round: 4 removed, 5 added. Added an address field to it has some context.

Reply:
No it's not. 

You are removing 5 balance related fields - `Available`, `Free`, `Reserved`, `Misc Frozen` and `Fee frozen` and replacing them with 4 balance related fields: `Transferable`, `Locked`, `Bonded` and `Total`. (`Address` is not related to balance so we don't talk about it). 

What I'm asking is what's the relationship between the 5 old fields and the 4 new ones ? Obviously one of them doesn't exist in the new implementation and I'm wondering which one and why that is.

Reply:
Ah good catch, now I see it. I removed `feeFrozen`, but that was not intentional. Now reading about each field, seems like the calculation for `transferable` should subtract both `miscFrozen` and `feeFrozen`.

Also seeing that balance calc is not that simple, will make some adjustments.

Reply:
Moved things around to match what Polkadot-JS is currently doing and properley calc each field. Will add a description to the PR.